docs/upgrading/upgrading-your-package.md

```diff
@@ -0,0 +1,542 @@
+# Upgrading your package to 1.0 APIs
+
+Atom is rapidly approaching 1.0. Much of the effort leading up to the 1.0 has been cleaning up APIs in an attempt to future proof, and make a more pleasant experience developing packages.
+
+This document will guide you through the large bits of upgrading your package to work with 1.0 APIs.
+
+## TL;DR
+
+We've set deprecation messages and errors in strategic places to help make sure you don't miss anything. You should be able to get 95% of the way to an updated package just by fixing errors and deprecations. There are a couple of things you need to do to enable all these errors and deprecations.
+
+### Use atom-space-pen-views
+
+If you use any class from `require 'atom'` with a `$` or `View` in the name, add the `atom-space-pen-views` module to your package's `package.json` file's dependencies:
+
+```js
+{
+  "dependencies": {
+    "atom-space-pen-views": "^2.0.2"
+  }
+}
+```
+
+Then run `apm install` in your package directory.
+
+### Require views from atom-space-pen-views
```

Comment:
Maybe it would be worth mentioning the need to rename the `afterAttach` and `beforeRemove` hooks in the section about switching views. Also might be worth mentioning that the `::subscribe` method is no longer provided and link to the section about subscriptions?